controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -60,7 +70,9 @@ type KubeadmControlPlaneReconciler struct {
 	Client client.Client
 	Log    logr.Logger
 
-	TemplateCloner TemplateCloner
+	TemplateCloner         TemplateCloner
+	KubeadmConfigGenerator KubeadmConfigGenerator
```

Comment:
Should KubeadmConfigGenerator and MachineGenerator be set in main.go the same way TemplateCloner is?

Reply:
Yes, this was the bug I hit late on Friday, thanks for catching that I didn't properly backport it to this PR :)

Reply:
Fixed